admission-controller/init/src/index.ts

```diff
@@ -0,0 +1,64 @@
+import k8s from '@kubernetes/client-node';
+import retry from 'async-retry';
+import logger, { formatLog } from './utils/logger.js';
+import { generateCertificates, isCertValid } from './utils/certificates.js';
+import { getSecretCertificate, applySecretCertificate, getWebhookConfiguration, patchWebhookCertificate } from './utils/kubernetes.js';
+
+const NAMESPACE = 'monokle-admission-controller';
+const SECRET_NAME = 'monokle-admission-controller-tls';
+const WEBHOOK_NAME = 'monokle-admission-controller-webhook';
+
+(async () => {
+  await retry(run, {
+    retries: 5,
+    factor: 3,
+    onRetry: (err, attempt) => {
+      logger.error(formatLog(`Cert init attempt ${attempt} failed. Retrying...`, err));
+    }
+  });
+})();
+
+// The flow is as follows:
+//
+// 1. Fetch webhook resource.
+//   - If there is none, it doesn't make sense to continue. Throw an error and retry.
+// 2. Fetch secret data (containing our cert).
+// 3. Check cert validity.
+//   - If valid, exit.
+//   - If empty or invalid, generate new secret.
```

Comment:
What would happen if the cert is about to expire? I suggest certificate renewal for this if it's not too much effort as otherwise there will be this point in time where it will fail until restart.

Reply:
My gut feeling is telling me to renew it after 2 months for a certification lasting 3 months. It will be a bit tricky though because the pod has to restart in that last month or it will no longer work. Can we maybe increase it to 6 month cert duration and renew after 3 months? That gives more leeway and is still more than secure enough.

Reply:
Good point, adjusted as suggested in https://github.com/kubeshop/monokle-admission-controller/pull/7/commits/47384f11e6d56e4b625d34767ca4301071a67333.